packages/headless/src/chat.ts

```diff
@@ -53,11 +61,18 @@ type State = {
    */
   fetching: Record<number, boolean | undefined>;
   /**
-   * Map from room id to error messages
+   * Map from room id to error messages (indicating problem with entire chat)
    */
   errors: Record<number, string | null | undefined>;
+  /**
+   * Map from room id to roomMessages to show the type of error that happened
+   * will be used to determine the type of retry function in the front end
+   */
+  messageErrors: Record<number, Record<number | string, ErrorType>>;
```

Comment:
Please make this type more specific. In stead of using non-specific types like numbers and strings.

Reply:
Why not including the errors with the messsage itself? why having it as a seperate map?